sbi/neural_nets/embedding_nets/causal_cnn.py

```diff
@@ -0,0 +1,176 @@
+# This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
+# under the Apache License Version 2.0, see <https://www.apache.org/licenses/>
+
+from typing import List, Optional, Tuple
+
+from torch import Tensor, nn
+
+from sbi.neural_nets.embedding_nets.cnn import calculate_filter_output_size
+
+
+def CausalCNN_1d_block(in_channels, out_channels, kernel_size, dilation, activation):
+    """Returns a block that computes a causal convolution by left padding the input
+
+    Args:
+        in_channels: number of input channels
+        out_channels: number of output channels wanted
+        kernel_size: wanted kernel size
+        dilation: dilation to use in the convolution,
+        activation: activation function to append to the convolution
+
+    Returns:
+        an nn.Sequential object that combines left-padding, 1D convolution
+        and the provided activation
+    """
+    padding_size = dilation * (kernel_size - 1)
+    padding = nn.ZeroPad1d(padding=(padding_size, 0))
+    conv_layer = nn.Conv1d(
+        in_channels=in_channels,
+        out_channels=out_channels,
+        kernel_size=kernel_size,
+        dilation=dilation,
+        stride=1,
+        padding=0,
+    )
+    return nn.Sequential(padding, conv_layer, activation)
+
+
+class CausalCNNEmbedding(nn.Module):
+    def __init__(
+        self,
+        input_shape: Tuple,
+        in_channels: int = 1,
+        out_channels_per_layer: Optional[List] = None,
+        dilation_per_layer: Optional[List] = None,
+        num_conv_layers: int = 5,
+        activation: nn.Module = nn.LeakyReLU(inplace=True),
+        pool_kernel_size: int = 160,
+        kernel_size: int = 2,
+        aggregator: Optional[nn.Module] = None,
+        output_dim: int = 20,
+    ):
+        """Embedding network that uses 1D causal convolutions
+        This is a simplified version of the architecture introduced for
+        the speech recognition task in the WaveNet paper (van den Oord, et al. (2016))
```

Comment:
please use a short description (up to one line) in the beginning, then a new paragraph with a more detailed description.